### PR TITLE
Adds a cop for validating against rails I18n

### PR DESCRIPTION
**Pull request by choubacha**
Many repos use rails for I18n so this commit adds a check to make sure
that sentence-like strings are wrapped in the translate function.

Ran this against the v3 codebase with exceptions being ignored and found 166 untranslated strings. Here were the config:

```yaml
RailsI18n/DecorateString:
  Enabled: true
  IgnoreExceptions: true
  Exclude:
    - '**/*.gemspec'
    - '**/*.rake'
    - '**/db/**/*'
```

### CODE REVIEW

File: lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          end
+
+          def dstr_contains_sentence?(node)
+            node.children.any? { |child| sentence?(child) }
+          end
+
+          def check_for_parent_decorator(node)
+            return if parent_is_translator?(node.parent)
+            return if parent_is_indexer?(node.parent)
+            return if ignoring_raised_parent?(node.parent)
```
Comment (joeltaylor) [inline lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb:72]: The named conditionals are super helpful 👍

**Review by joeltaylor — APPROVED** [review]
Nice work! Spec coverage is 💯

**Review by cschlessinger — COMMENTED** [review]
I'm not sure I understand everything the `DecorateString` methods are doing so not sure I can approve but the spec was really neat, I learned some cool stuff!

File: lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb (hunk 1 above)
Comment (shin-taskrabbit) [reply lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb:72]: Agree. I love this style a lot.  Still learning RuboCop though.

File: lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            check_for_parent_decorator(node)
+          end
+
+          private
+
+          def sentence?(node)
+            child = node.children[0]
+            if child.is_a?(String)
+              if child.valid_encoding?
+                child.encode(Encoding::UTF_8).chomp =~ STRING_REGEXP
```
Comment (shin-taskrabbit) [inline lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb:54]: Due to my lack of knowledge, I am trying some out in IRB but I shouldn't be blocking you from submitting (Joel already approved it so no block here).  Does this cover string interpolation case or that is outside of scope? I am just asking.

**Review by shin-taskrabbit — APPROVED** [review]

File: lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb (hunk 2 above)
Comment (choubacha) [reply lib/rubocop/cop/i18n/rails_i18n/decorate_string.rb:54]: `dstr` (from what Joe and I read) are interpolated strings, so it should work for them without issue. I can add a test case for it though. However, this code was more or less just copied from the original GetText implementation.

**Pull request merged**